File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        res = 0
        n = len(nums)
        l = 0
        freqs = {}
        for r in range(n):
            freqs[nums[r]] = freqs.get(nums[r], 0) + 1
            while freqs[nums[r]] > k:
                freqs[nums[l]] -= 1
                l += 1
            res = max(res, r - l + 1)
        return res

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        int res = 0;
        int n = nums.size();
        int l = 0;
        unordered_map<int, int> freqs;
        for(int r = 0; r < n; r++)
        {
            freqs[nums[r]]++;
            while(freqs[nums[r]] > k)
            {
                freqs[nums[l]]--;
                l++;
            }
            res = max(res, r - l + 1);
        }
        return res;
    }
};